From 24715768b0b6f2288c1b46ed13f417fa23677759 Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Fri, 20 Dec 2019 07:30:10 -0600
Subject: [PATCH 38/43] arm64: dts: logicpd-imx8mm-kit: Enable SPI EEPROM

The Phoenix baseboard has an SPI EEPROM, but the chip select is
shared with the UART, so this patch removes the hardware handshaking
muxing, and reassigns the GPIO pin to the chip select for the ECSPI2.

The at25 mode was added with Nathan Kro's guidance to configure the
pagesize, size and address-width.

Signed-off-by: Adam Ford <aford173@gmail.com>
---
 .../dts/freescale/lpd-imx8mm-baseboard.dtsi   | 31 +++++++++++++++++--
 1 file changed, 28 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/lpd-imx8mm-baseboard.dtsi b/arch/arm64/boot/dts/freescale/lpd-imx8mm-baseboard.dtsi
index ff6ff2d08f2f..cd7869015e0d 100644
--- a/arch/arm64/boot/dts/freescale/lpd-imx8mm-baseboard.dtsi
+++ b/arch/arm64/boot/dts/freescale/lpd-imx8mm-baseboard.dtsi
@@ -144,6 +144,15 @@
 			>;
 		};
 
+		pinctrl_espi2: espi2grp {
+			fsl,pins = <
+				MX8MM_IOMUXC_ECSPI2_SCLK_ECSPI2_SCLK		0x82
+				MX8MM_IOMUXC_ECSPI2_MOSI_ECSPI2_MOSI		0x82
+				MX8MM_IOMUXC_ECSPI2_MISO_ECSPI2_MISO		0x82
+				MX8MM_IOMUXC_ECSPI1_SS0_GPIO5_IO9		0x41
+			>;
+		};
+
 		pinctrl_led3: led3grp {
 			fsl,pins = <
 				MX8MM_IOMUXC_SAI3_RXFS_GPIO4_IO28	0x41
@@ -203,8 +212,6 @@
 			fsl,pins = <
 				MX8MM_IOMUXC_ECSPI1_SCLK_UART3_DCE_RX		0x140
 				MX8MM_IOMUXC_ECSPI1_MOSI_UART3_DCE_TX		0x140
-				MX8MM_IOMUXC_ECSPI1_SS0_UART3_DCE_RTS_B		0x140
-				MX8MM_IOMUXC_ECSPI1_MISO_UART3_DCE_CTS_B	0x140
 			>;
 		};
 
@@ -263,6 +270,25 @@
 	};
 };
 
+&ecspi2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_espi2>;
+	status = "okay";
+	cs-gpios = <&gpio5 9 0>;
+
+	at25@0 {
+		compatible = "atmel,at25";
+		reg = <0>;
+		spi-max-frequency = <5000000>;
+		spi-cpha;
+		spi-cpol;
+
+		pagesize = <32>;
+		size = <2048>;
+		address-width = <16>;
+	};
+};
+
 &i2c2 {
 	clock-frequency = <400000>;
 	pinctrl-names = "default";
@@ -402,7 +428,6 @@
 	pinctrl-0 = <&pinctrl_uart3>;
 	assigned-clocks = <&clk IMX8MM_CLK_UART3>;
 	assigned-clock-parents = <&clk IMX8MM_SYS_PLL1_80M>;
-	fsl,uart-has-rtscts;
 	status = "okay";
 };
 
-- 
2.17.1

